lib/bforest/README.md

```diff
@@ -0,0 +1,2 @@
+This crate contains array-based data structures used by the core Cranelift code
+generator which represent a set of small ordered sets or maps.
```

Comment:
It'd be good to add some of the content from the top-level comment in the current mod.rs file about how this code is specialized.